asm-utils/src/data_storage.rs

```diff
@@ -7,23 +7,23 @@ use crate::{
     utils::{alignment_size, next_aligned},
 };
 
-pub enum SingleDataValue<'a> {
+pub enum SingleDataValue {
     Value(u32),
-    LabelReference(&'a str),
-    Offset(&'a str, &'a str),
+    LabelReference(String),
+    Offset(String, String),
```

Comment:
This change won't make new String copies, the original Strings are being moved now.

Reply:
nice